### PR TITLE
Add Blood Cult Cat

### PR DESCRIPTION
**Pull request by ErhardSteinhauer**
## About the PR
- Added blood cult cat, pet ghost role (whitelisted). Named after towns that hosted famous witch trials in the past.
- Added blood cult cat to the Blood Moon bluespace event.
- Has a small chance of spawning on vgroids.

Abilities:
- Toggleable red light.
- Dissonant whispers.
- Eldritch Push: slow moving projectile, that deals 100 stamina damage. 5 minutes cooldown.

## Changes to dev_map:
- Added Scrap Processor with scrap chunks, Blueprint Lithograph and hacked mercenary techfab.
- Added blood cult cat spawner.
- Added turbolaser mk3, holopickaxe and holoflare pistol to weapon pile, holograffiti projector to tools pile

TODO:
- [x] Modest abilities for ghost to entertain themselves and those around them.
- [x] Spawns.

## Why / Balance
Cats!

## Media
image
image

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**
:cl: erhardsteinhauer
- add: Added the blood cult cat ghost role (for players to pledge their allegiance to).

### CODE REVIEW

File: Resources/Prototypes/_NF/Datasets/Names/cat_cult.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+- type: dataset # Names of town that held witch trials
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Datasets/Names/cat_cult.yml:1]: For your perusal: https://en.wikipedia.org/wiki/List_of_people_executed_for_witchcraft#List_of_people_executed_for_witchcraft

Glamis, Waterhouse, Leatherlips, Wardwell and maybe "Soulmother" seem like interesting names to me.

File: Resources/Prototypes/_NF/Entities/Mobs/NPCs/pets.yml [hunk 2]
```diff
@@ -223,6 +223,106 @@
...
+  description: "Summon forth an eldritch hand. [color=yellow]3 minutes cooldown.[/color]"
+  components:
+  - type: WorldTargetAction
+    icon: { sprite : _NF/Effects/bloodcultprojectiles.rsi, state: hand }
+    event: !type:ActionGunShootEvent
+    priority: 2
+    checkCanAccess: false
+    range: 0
+    itemIconStyle: BigAction
+    useDelay: 300
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Mobs/NPCs/pets.yml:324]: The delay here doesn't seem to be working on my end, I'd expect a cooldown ring, but the baby dragon's also isn't working.  Might've been broken, unsure why.

**Review by whatston3 — COMMENTED** [review]
Neat stuff - left a few comments about the cat's usage, but seems like a nice addition to the blood cult.

File: Resources/Prototypes/_NF/Entities/Mobs/NPCs/pets.yml (hunk 2 above)
Comment (ErhardSteinhauer) [reply Resources/Prototypes/_NF/Entities/Mobs/NPCs/pets.yml:323]: Yeah, it doesn't

Comment (whatston3) [reply Resources/Prototypes/_NF/Entities/Mobs/NPCs/pets.yml:323]: Alrighty, noted.

**Review by whatston3 — APPROVED** [review]
Pushed a small set of changes - rule description spelling fix, no "parent: invalid", clarpy & blood cult cat crit/dead sprite use the live ones (hats look odd otherwise).  Seems good, give it a look when you can.

File: Resources/Prototypes/_NF/Procedural/chromite_vgroid.yml [hunk 3]
```diff
@@ -272,12 +272,12 @@
...
-    maxCount: 1
-    groups:
-    - id: SpawnMobCatBloodCultLowChance
-      amount: 1
+#  - !type:MobsDunGen # Cat dies to space, big sad
+#    minCount: 1
+#    maxCount: 1
+#    groups:
+#    - id: SpawnMobCatBloodCultLowChance
+#      amount: 1
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Procedural/chromite_vgroid.yml:280]: Could either remove the respirator (it's a magic spooky cat, mistake could probably work the same way) or give it internals via a loadout.

File: Resources/Prototypes/_NF/Entities/Mobs/NPCs/pets.yml [hunk 4]
```diff
@@ -332,7 +324,7 @@
 # Adopted from space-wizards #26265 (Thank you, PolterTzi)
 - type: entity
   name: Yip Yip
-  parent: MobKobold
+  parent: [NFMobPet, MobKobold]
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Mobs/NPCs/pets.yml:327]: Are we doing this to Pun Pun too?

**Pull request merged**